framework/src/controller/ws/ws_server.ts

```diff
@@ -72,12 +79,24 @@ export class WSServer {
 			});
 		}
 
-		this.server.on('connection', socket =>
-			this._handleConnection(socket as WebSocketWithTracking, messageHandler),
-		);
+		this.server.on('connection', (socket, req) => {
+			const { origin } = req.headers;
+			if (
```

Comment:
shouldnt we reject if origin is not set && _accessControlAllowOrigin !== '*'?

Reply:
Idk but that wouldn't that be an ambiguous situation ? I am not aware how origin is set but if you think it could be a security issue and it's safe to do it I'll add this condition.

Reply:
origin is `undefined` in case of non browser or same origin and for the non browser environment, origin doesn't probably help much regarding the security
so i guess we can keep it as it is